src/main/java/seedu/address/logic/commands/editcommand/EditMentorCommand.java

```diff
@@ -26,25 +40,49 @@ public EditMentorCommand(Id id, EditMentorDescriptor editMentorDescriptor) {
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
 
-        // see EditIssueCommand
+        Mentor mentorToEdit;
+        try {
+            mentorToEdit = model.getMentor(this.id);
+        } catch (AlfredException e) {
+            throw new CommandException(MESSAGE_INVALID_MENTOR_DISPLAYED_INDEX);
+        }
+        Mentor editedMentor = this.createEditedMentor(mentorToEdit,
+                this.editMentorDescriptor);
 
-        return new CommandResult("");
+        // Model should check if there exists duplicates in list
+        /*
+         * i.e
+         * if (!personToEdit.isSamePerson(editedPerson) && model.hasPerson(editedPerson)) {
+         *     throw new CommandException(MESSAGE_DUPLICATE_PERSON);
+         * }
+         */
+        if (model.updateMentor(this.id, editedMentor)) {
```

Comment:
Nitpicking here: Would prefer this as the main path should be the correct one and the if statements should catch the edge cases
```suggestion
        if (!model.updateMentor(this.id, editedMentor)) {
            // handle edge case
         }
         return new CommndResult(Success)
```

Reply:
Good point. Will change